staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -2218,3 +2219,41 @@ func TestThrottledLogger(t *testing.T) {
 		t.Fatalf("expected %v log messages, but got %v", e, a)
 	}
 }
+
+func TestRequestMaxRetries(t *testing.T) {
+	successAfterRetry := 5
+	retried := 0
+	retryOneTimeHandler := func(w http.ResponseWriter, req *http.Request) {
+		if retried >= successAfterRetry {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+		w.Header().Set("Retry-After", "1")
+		w.WriteHeader(http.StatusTooManyRequests)
+		retried++
+	}
+	testServer := httptest.NewServer(http.HandlerFunc(retryOneTimeHandler))
+	defer testServer.Close()
+
+	u, err := url.Parse(testServer.URL)
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = NewRequestWithClient(u, "", defaultContentConfig(), testServer.Client()).
+		Verb("get").
+		MaxRetries(0).
```

Comment:
How anout making this `successAfterRetry-1` instead of 0?  That would make for a sharper test.

Reply:
I suggest:
* keep this case
* add the case Mike suggests
* reset `retried` prior to each and verify its value after each case. Also do this for the success case below.

Reply:
ah.. great catch. by adding the case mike suggested, i found that the original behavior is actually retrying 9 times instead of 10. a sketch of the original implementation:

```go
retries := 0
client.Do(..)
retries++
if retries < r.maxRetries {  //  should be `<=` instead
  sleep()
}
```